traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices_required_capabilities.go

```diff
@@ -219,22 +219,51 @@ func (rc *RequiredCapability) getCapabilities(tenantIDs []int) ([]tc.DeliverySer
 // Delete implements the api.CRUDer interface.
 func (rc *RequiredCapability) Delete() (error, error, int) {
 	authorized, err := rc.isTenantAuthorized()
-	if err != nil {
-		return nil, errors.New("checking tenant: " + err.Error()), http.StatusInternalServerError
-	} else if !authorized {
+	if !authorized {
 		return errors.New("not authorized on this tenant"), nil, http.StatusForbidden
+	} else if err != nil {
+		return nil, fmt.Errorf("checking authorization for existing DS ID: %s" + err.Error()), http.StatusInternalServerError
+	}
+
+	// Check if the Delivery Service is associated with the Required Capability
+	statusCode, err := rc.exists()
+	if err != nil {
+		return err, nil, statusCode
 	}
 
 	return api.GenericDelete(rc)
 }
 
+// exists checks to see if a required capability is assigned to a delivery service
+func (rc *RequiredCapability) exists() (int, error) {
```

Comment:
Is this not caught by https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/api/generic_crud.go#L169? is this just to clean up the error message?

Reply:
I dont have my head wrapped around all of these interfaces yet....

this was added before i realized the auth if check needed switched. thanks for pointing out the functionality of this interface. removing this code. updates coming